builtin/logical/aws/secret_access_keys.go

```diff
@@ -111,6 +111,48 @@ func (b *backend) secretTokenCreate(s logical.Storage,
 	return resp, nil
 }
 
+func (b *backend) assumeRole(s logical.Storage,
+	displayName, policyName, policy string,
+	lifeTimeInSeconds *int64) (*logical.Response, error) {
```

Comment:
Passing a pointer to an `int64` is strange in Go-land.You can just do `&lifeTimeInSeconds` in the `AssumeRoleInput` struct. According to the docs, "the value can range from 900 seconds (15 minutes) to 3600 seconds (1 hour). By default, the value is set to 3600 seconds." I'm not sure what it would do to a default-zero `int64` (which may not be the same as what it *should* do), but if you're concerned about that, check ahead of time to see if it's zero and set the final value to nil if so.